### PR TITLE
docs - awscc_iam_role

### PR DESCRIPTION
**Pull request by naveen-amzn**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

Relates OR Closes #0000

### CODE REVIEW

File: examples/resources/awscc_iam_role/iam_role_with_inline.tf [hunk 1]
```diff
@@ -0,0 +1,53 @@
+data "aws_iam_policy_document" "instance_aeessume_role_policy" {
```
Comment (rahul-amzn) [inline examples/resources/awscc_iam_role/iam_role_with_inline.tf:1]: seems typo - instance_aeessume_role_policy ?

File: docs/resources/iam_role.md [hunk 2]
```diff
@@ -10,6 +9,193 @@ description: |-
...
+        Sid    = ""
+        Principal = {
+          Service = "ec2.amazonaws.com"
+        }
+      },
+    ]
+  })
+  max_session_duration = 7200
+  path                 = "/"
+  tags = [{
```
Comment (rahul-amzn) [inline docs/resources/iam_role.md:35]: @naveen-amzn may add  

```
  tags = [{
    key   = "Modified By"
    value = "AWSCC"
  }]
```

File: docs/resources/iam_role.md [hunk 3]
```diff
@@ -10,6 +9,193 @@ description: |-
...
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+
+data "aws_iam_policy_document" "sample_inline_1" {
+  statement {
+    sid       = "tes2t"
```
Comment (rahul-amzn) [inline docs/resources/iam_role.md:96]: typo tes2t ?

File: docs/resources/iam_role.md [hunk 4]
```diff
@@ -10,6 +9,193 @@ description: |-
...
+  path                        = "/"
+  policies = [{
+    policy_document = data.aws_iam_policy_document.sample_inline_1.json
+    policy_name     = "fist_inline_policy"
+    },
+    {
+      policy_document = data.aws_iam_policy_document.sample_inline_2.json
+      policy_name     = "second_inline_policy"
+  }]
+  tags = [{
```
Comment (rahul-amzn) [inline docs/resources/iam_role.md:126]: same comment for tags (everywhere)

File: examples/resources/awscc_iam_role/iam_role_with_inline.tf (hunk 1 above)
Comment (naveen-amzn) [reply examples/resources/awscc_iam_role/iam_role_with_inline.tf]: Fixed typo.

File: examples/resources/awscc_iam_role/iam_role_with_managed.tf [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+
+resource "awscc_iam_role" "main" {
+  role_name                   = "sample_iam_role"
+  description                 = "This is a sample IAM role"
+  assume_role_policy_document = data.aws_iam_policy_document.instance_assume_role_policy.json
+  managed_policy_arns         = [aws_iam_policy.policy_one.arn, aws_iam_policy.policy_two.arn]
+  path                        = "/"
+  tags = [{
+    key   = "NAME"
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_role/iam_role_with_managed.tf:50]:
```suggestion
    key   = "Name"
```

File: examples/resources/awscc_iam_role/iam_role_with_inline.tf [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  policies = [{
+    policy_document = data.aws_iam_policy_document.sample_inline_1.json
+    policy_name     = "fist_inline_policy"
+    },
+    {
+      policy_document = data.aws_iam_policy_document.sample_inline_2.json
+      policy_name     = "second_inline_policy"
+  }]
+  tags = [{
+    key   = "NAME"
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_role/iam_role_with_inline.tf:46]:
```suggestion
    key   = "Name"
```

File: examples/resources/awscc_iam_role/iam_role_with_assume_role.tf [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+
+resource "awscc_iam_role" "main" {
+  role_name                   = "sample_iam_role"
+  description                 = "This is a sample IAM role"
+  assume_role_policy_document = data.aws_iam_policy_document.instance_assume_role_policy.json
+  path                        = "/"
+  tags = [{
+    key   = "NAME"
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_role/iam_role_with_assume_role.tf:19]:
```suggestion
    key   = "Name"
```

File: examples/resources/awscc_iam_role/iam_role.tf [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        Principal = {
+          Service = "ec2.amazonaws.com"
+        }
+      },
+    ]
+  })
+  max_session_duration = 7200
+  path                 = "/"
+  tags = [{
+    key   = "NAME"
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_role/iam_role.tf:20]:
```suggestion
    key   = "Name"
```

File: examples/resources/awscc_iam_role/iam_role.tf [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    value = "Sample-IAM-Role"
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_role/iam_role.tf:21]:
```suggestion
    value = "Sample IAM Role"
```

File: examples/resources/awscc_iam_role/iam_role.tf [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    },
+    {
+      key   = "Environment"
+      value = "development"
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_role/iam_role.tf:25]:
```suggestion
      value = "Development"
```

File: examples/resources/awscc_iam_role/iam_role_with_assume_role.tf [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    value = "Sample-IAM-Role"
+    },
+    {
+      key   = "Environment"
+      value = "development"
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_role/iam_role_with_assume_role.tf:24]:
```suggestion
      value = "Development"
```

File: examples/resources/awscc_iam_role/iam_role_with_inline.tf [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+    value = "Sample-IAM-Role"
+    },
+    {
+      key   = "Environment"
+      value = "development"
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_role/iam_role_with_inline.tf:51]:
```suggestion
      value = "Development"
```

File: examples/resources/awscc_iam_role/iam_role_with_managed.tf [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+    value = "Sample-IAM-Role"
+    },
+    {
+      key   = "Environment"
+      value = "development"
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_role/iam_role_with_managed.tf:55]:
```suggestion
      value = "Development"
```

File: examples/resources/awscc_iam_role/iam_role_with_permission_boundry.tf [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+
+resource "awscc_iam_role" "main" {
+  role_name                   = "sample_iam_role"
+  description                 = "This is a sample IAM role"
+  assume_role_policy_document = data.aws_iam_policy_document.instance_assume_role_policy.json
+  managed_policy_arns         = [aws_iam_policy.policy_one.arn]
+  permissions_boundary        = aws_iam_policy.s3_permission_boundry_policy.arn
+  path                        = "/"
+  tags = [{
+    key   = "NAME"
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_role/iam_role_with_permission_boundry.tf:52]:
```suggestion
    key   = "Name"
```

File: examples/resources/awscc_iam_role/iam_role_with_permission_boundry.tf [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+    value = "Sample-IAM-Role"
+    },
+    {
+      key   = "Environment"
+      value = "development"
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_role/iam_role_with_permission_boundry.tf:57]:
```suggestion
      value = "Development"
```

File: examples/resources/awscc_iam_role/iam_role_with_permission_boundry.tf [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+      {
+        Action   = ["s3:ListAllMyBuckets", "s3:ListBucket", "s3:HeadBucket"]
+        Effect   = "Allow"
+        Resource = "*"
+      },
+    ]
+  })
+}
+
+resource "aws_iam_policy" "s3_permission_boundry_policy" {
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_role/iam_role_with_permission_boundry.tf:28]: typo for boundary, please check across all resources

**Review by wellsiau-aws — CHANGES_REQUESTED** [review]

**Review by wellsiau-aws — APPROVED** [review]
LGTM

**Comment by ewbankkit** [comment]
@naveen-amzn Thanks for the contribution :tada: :clap:.

**Pull request merged**